File: src/main/java/net/guizhanss/ultimategenerators2/implementation/listeners/TranslationsLoadListener.java
package net.guizhanss.ultimategenerators2.implementation.listeners;

import javax.annotation.Nonnull;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

import net.guizhanss.slimefuntranslation.api.events.TranslationsLoadEvent;
import net.guizhanss.ultimategenerators2.UltimateGenerators2;

public final class TranslationsLoadListener implements Listener {
    public TranslationsLoadListener(@Nonnull UltimateGenerators2 plugin) {
        if (UltimateGenerators2.getIntegrationService().isSlimefunTranslationEnabled()) {
            plugin.getServer().getPluginManager().registerEvents(this, plugin);
        }
    }

    @EventHandler
    public void onLoad(@Nonnull TranslationsLoadEvent e) {
        UltimateGenerators2.getIntegrationService().loadTranslations();
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/implementation/listeners/TranslationsLoadListener.kt
package net.guizhanss.ultimategenerators2.implementation.listeners

import net.guizhanss.slimefuntranslation.api.events.TranslationsLoadEvent
import net.guizhanss.ultimategenerators2.UltimateGenerators2
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class TranslationsLoadListener(val plugin: UltimateGenerators2) : Listener {
    init {
        if (UltimateGenerators2.integrationService().slimefunTranslationEnabled) {
            plugin.server.pluginManager.registerEvents(this, plugin)
        }
    }

    @EventHandler
    fun onLoad(e: TranslationsLoadEvent) {
        UltimateGenerators2.integrationService().loadTranslations()
    }
}
